### ISSUE TITLE
'use-after-dtor' issue in StaticDecoder.cpp

### ISSUE THREAD

**Issue by pavik** [Refinitiv/Real-Time-SDK#275]
Issue appears in EMA library when decoding JSON payload in [StaticDecoder::setRsslData](https://github.com/Refinitiv/Real-Time-SDK/blob/master/Cpp-C/Ema/Src/Access/Impl/StaticDecoder.cpp#L73)
Callstack:
```
void StaticDecoder::setRsslData( Data* pData, RsslBuffer* pRsslBuffer, RsslDataType rsslType, UInt8 majVer, UInt8 minVer, const RsslDataDictionary* dictionary )
{
     ...
     morph( pData, dType );   // calls destructor on pData
     if ( !pData->getDecoder().setRsslData( majVer, minVer, pRsslBuffer, dictionary, 0 ) )  // <- pure virt function call leads to crash
     ...
}
...
void StaticDecoder::morph( Data* data, DataType::DataTypeEnum dType )
{
	if ( data->getDataType() != dType )
	{
             data->~Data();  // <- object destroyed
             StaticDecoder::create( data, dType );    // <- switch/case does not have default branch so object remains destroyed
	}
}
```

Hope I explained well, feel free to ask more details if needed.

**Comment by ViktorYelizarov** [comment]
@pavik  Thank you for bringing this issue to our attention! We created an internal Jira to investigate it.

**Comment by ViktorYelizarov** [comment]
@pavik Could you provide an example how to reproduce this issue?

**Comment by pavik** [comment]
Unfortunately I can't provide code snippet it is proprietary. In a nutshell, ETA provider sends refresh message with json payload and EMA consumer initiate request. btw, there is no json data type in EMA DataTypeEnum, it is detected as NoDataEnum. Fortunately payload can be read via `msg.getPayload().getData().getAsHex().c_buf()`. imho, anyway better to fix, it is dangerous code path that can leave object in destructed state.

**Comment by vlevendel** [comment]
Fixed with version Real-Time-SDK-2.2.3.L1. Please reopen if there are additional concerns.
